src/commands/build.ts

```diff
@@ -49,67 +56,157 @@ export const build: CommandModule<CliGlobalOptions, CliCommandOptions> = {
+  const templateMode = !!template;
+
+  const buildOptions: BuildAndUploadOptions = {
+    dir,
+    contentProvider,
+    uploadTo,
+    userTimeout,
+    skipSave,
+    skipUpload,
+    composeFileName,
+    requireGitData,
+    deleteOldPins,
+  };
+
+  const verbosityOptions: VerbosityOptions = { renderer: verbose ? "verbose" : silent ? "silent" : "default" }
+
+  const buildTasks = templateMode ?
+    handleTemplateBuild({ buildOptions, variantsDir, variants, verbosityOptions }) :
+    handleSinglePkgBuild({ buildOptions, verbosityOptions });
+
+  return Promise.all(buildTasks.map((task) => task.run()));
+}
+
+function handleTemplateBuild({
+  buildOptions,
+  variantsDir,
+  variants,
+  verbosityOptions
+}: {
+  buildOptions: BuildAndUploadOptions;
+  variantsDir: string;
+  variants?: string;
+  verbosityOptions: VerbosityOptions;
+}
+): Listr<ListrContextBuildAndPublish>[] {
+  const variantsDirPath = path.join(buildOptions.dir, variantsDir);
+  const variantNames = getVariantNames({ variantsDirPath, variants, templateMode: true });
+
+  console.log(`${chalk.dim(`Building package from template for variants ${variants}...`)}`);
+
+  return variantNames.map((variantName) => new Listr(
+    buildAndUpload({ ...buildOptions, variantName, variantsDirPath }),
+    verbosityOptions
+  ));
 }
+
+function handleSinglePkgBuild({
+  buildOptions,
+  verbosityOptions
+}: {
+  buildOptions: BuildAndUploadOptions;
+  verbosityOptions: VerbosityOptions;
+}): Listr<ListrContextBuildAndPublish>[] {
+  console.log(`${chalk.dim(`Building single package...`)}`);
+
+  return [new Listr(
+    buildAndUpload(buildOptions),
+    verbosityOptions
+  )];
+}
+
+/**
+ * Retrieves the valid variant names based on the specified variants and the available directories.
+ * 
+ * @param {string} variantsDirPath - The path to the directory containing variant directories.
+ * @param {string | undefined} variants - A comma-separated string of specified variant names.
+ * @param {boolean} templateMode - Flag indicating if the template mode is enabled.
+ * @returns {string[]} An array of valid variant names.
+ */
+function getVariantNames({ variantsDirPath, variants, templateMode }: { variantsDirPath: string; variants?: string; templateMode: boolean; }): string[] {
+  if (!templateMode) return [];
+
+  const allVariantNames = getAllDirectoryNamesInPath(variantsDirPath);
+  if (!variants) return allVariantNames; // If no specific variants are provided, use all available directories.
+
+  const specifiedVariantNames = variants.split(",").map(name => name.trim()); // Split and trim the specified variants to ensure clean comparisons.
+  const validVariantNames = specifiedVariantNames.filter(name => allVariantNames.includes(name)); // Filter out invalid names.
+
+  if (validVariantNames.length !== specifiedVariantNames.length) {
+    console.warn(chalk.yellow(`Warning: Some specified variants are not valid and will be ignored. Check the variants: ${variants}`));
```

Comment:
I consider it dangerous not to throw an error here. it might end with a non expected result by the developer